src/Toolkit.Forms/OverviewMap/OverviewMap.cs

```diff
@@ -0,0 +1,219 @@
+
+        private MapView? _overviewMapView;
+
+        static OverviewMap()
+        {
+            string template = @"<ControlTemplate xmlns=""http://xamarin.com/schemas/2014/forms""
+                                                 xmlns:x=""http://schemas.microsoft.com/winfx/2009/xaml""
+                                                 xmlns:esri=""clr-namespace:Esri.ArcGISRuntime.Xamarin.Forms;assembly=Esri.ArcGISRuntime.Xamarin.Forms""
+                                                 xmlns:internal=""clr-namespace:Esri.ArcGISRuntime.Toolkit.Xamarin.Forms.Internal;assembly=Esri.ArcGISRuntime.Toolkit.Xamarin.Forms"">
+                                    <Grid>
+                                        <Grid.Resources>
+                                            <internal:LoadStatusToVisibilityConverter x:Key=""LoadStatusToVisibilityConverter"" />
+                                        </Grid.Resources>
+                                        <ActivityIndicator IsRunning=""{Binding Source={x:Reference PART_MapView}, Path=Map.LoadStatus, Converter={StaticResource LoadStatusToVisibilityConverter}, ConverterParameter='Loading'}"" />
+                                        <Label Text=""Map failed to load. Did you forget an API key?"" IsVisible=""{Binding Source={x:Reference PART_MapView}, Path=Map.LoadStatus, Converter={StaticResource LoadStatusToVisibilityConverter}, ConverterParameter='FailedToLoad'}""  />
+                                        <esri:MapView x:Name=""PART_MapView"" IsVisible=""{Binding Source={x:Reference PART_MapView}, Path=Map.LoadStatus, Converter={StaticResource LoadStatusToVisibilityConverter}, ConverterParameter='Loaded'}"" />
+                                    </Grid>
+                                </ControlTemplate>";
+            DefaultControlTemplate = Extensions.LoadFromXaml(new ControlTemplate(), template);
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="OverviewMap"/> class.
+        /// </summary>
+        public OverviewMap()
+        {
+            ControlTemplate = DefaultControlTemplate;
+        }
+
+        /// <summary>
+        /// <inheritdoc />
+        /// </summary>
+        protected override void OnApplyTemplate()
+        {
+            base.OnApplyTemplate();
+
+            _overviewMapView = GetTemplateChild("PART_MapView") as MapView;
+
+            if (_overviewMapView != null)
+            {
+                _controller = new OverviewMapController(_overviewMapView)
+                {
+                    ScaleFactor = ScaleFactor,
+                    PointSymbol = PointSymbol,
+                    AreaSymbol = AreaSymbol,
+                    AttachedView = GeoView,
+                };
+                _overviewMapView.IsAttributionTextVisible = false;
+
+                if (Map == null)
+                {
+                    Map = new Map(BasemapStyle.ArcGISTopographic);
+                }
```

Comment:
This should be done in the constructor instead. Or set directly on the `_overviewMapView.Map` property so the public property stays `null`. Also you should probably set _overviewMapView.Map regardless, or it won't get applied if you set the property before applying the template